File: Data Structures & Algorithms/grumpy-bookstore-owner/submission-1.java
class Solution {
    public int maxSatisfied(int[] customers, int[] grumpy, int minutes) {
        int sum=0;
        int maxi=Integer.MIN_VALUE;
        for(int i=0;i<customers.length;i++){
            if(grumpy[i]==0){
            sum=sum+customers[i];
            }
            maxi=Math.max(maxi,sum);

        }
        for(int i=0;i<minutes;i++){
            if(grumpy[i]==1){
                sum=sum+customers[i];
                maxi=Math.max(maxi,sum);
            }
        }
        for(int i=minutes;i<customers.length;i++){
            if(grumpy[i]==1){sum=sum+customers[i];}
            if(grumpy[i-minutes]==1){sum=sum-customers[i-minutes];}
            maxi=Math.max(maxi,sum);

        }
        return maxi;
    }
}

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.py
class Solution:
    def maxi(self, arr: List[int],i: int ,j: int):
        maxi=0
        for k in range(i,j):
            if maxi< arr[k]:
                maxi=arr[k]
        return maxi


    def replaceElements(self, arr: List[int]) -> List[int]:
        for i in range(0,len(arr)):
            arr[i]=self.maxi(arr,i+1,len(arr))
        arr[len(arr)-1]=-1

        return arr

        